### PR TITLE
Search input component

### LINKED ISSUE #61: Search Input Reusable Component
Make Search Input separated reusable component

Search Input behavior + styles -> [**Wiki link**](https://github.com/ita-social-projects/what-front/wiki/Search-Input-separated-reusable-component)

Examples
image
image
image

### PR DESCRIPTION
**Pull request by PavelKundenko**
## Summary of [issue](https://github.com/ita-social-projects/what-front/issues/61)
Mock: 
image

[Wiki](https://github.com/ita-social-projects/what-front/wiki/Search-Input-separated-reusable-component)

This PR
closes #61 

## Summary of change

  - **onSearch**: ` func` - search button click handler, should accept parameter _inputValue_
  - **placeholder**: `string` - input placeholder
> Accepts 2 optional props: 
  - **className**: `string` - additional class that you can apply search container Default: _''_

### Examples: 
```
<Search
   onSearch={handleSearch}
   placeholder="Search something"
 />
```
image

### CODE REVIEW

File: src/components/search/search.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import Icon from '../../icon.js';
+import styles from './search.module.scss';
+
+export const Search = ({
+  onSearch,
+  onChange,
+  inputRef,
+  placeholder,
+  additionalCssClasses,
```
Comment (sarhan-azizov) [inline src/components/search/search.js:13]: ?
```suggestion
  className,
```

File: src/components/search/search.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+}) => (
+  <div className={classNames(
+    styles.search,
+    { [additionalCssClasses.join()]: additionalCssClasses.length },
+  )}
+  >
+    <input
+      type="text"
+      className={styles.search__input}
```
Comment (sarhan-azizov) [inline src/components/search/search.js:22]: styles.search__input

styles.searchInput

styles.field

File: src/components/search/search.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      ref={inputRef}
+    />
+    <div onClick={onSearch} className={styles.search__button}>
+      <Icon icon="Search" size={32} viewBox="0 0 50 50" />
+    </div>
+  </div>
+);
+
+Search.propTypes = {
+  onSearch: PropTypes.func.isRequired,
```
Comment (sarhan-azizov) [inline src/components/search/search.js:34]:
```suggestion
  onSearch: PropTypes.func.isRequired,
```

func.isRequired,

File: src/components/search/search.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  inputRef: PropTypes.shape({
+    current: PropTypes.oneOf([undefined]),
+  }).isRequired,
+  placeholder: PropTypes.string.isRequired,
+  onChange: PropTypes.func,
+  additionalCssClasses: PropTypes.arrayOf(PropTypes.string),
+};
+
+Search.defaultProps = {
+  onChange: () => false,
```
Comment (sarhan-azizov) [inline src/components/search/search.js:44]:
```suggestion
  onChange:undefined,
```

File: src/features/counter/counter.js [hunk 5]
```diff
@@ -1,24 +1,31 @@
...
+  const handleSubmitModal = () => {
     console.log('Modal submitted, something happened!');
-    closeModalHandler();
+    handleCloseModal();
+  };
+
+  const searchRef = useRef();
+
+  const handleSearch = () => {
+    console.log(`Search value = ${searchRef.current.value}`);
```
Comment (sarhan-azizov) [inline src/features/counter/counter.js:28]: this.props.onSearch(searchRef.current.value)

File: src/components/search/search.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import Icon from '../../icon.js';
+import styles from './search.module.scss';
+
+export const Search = ({
+  onSearch,
+  onChange,
+  inputRef,
```
Comment (sarhan-azizov) [inline src/components/search/search.js:11]: ?

File: src/index.html [hunk 7]
```diff
@@ -4,6 +4,7 @@
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.5.3/dist/css/bootstrap.min.css" integrity="sha384-TX8t27EcRE3e/ihU7zmQxVncDAy5uIKz4rEkgIXeMed4M0jlfIDPvg6uqKI2xXr2" crossorigin="anonymous">
```
Comment (sarhan-azizov) [inline src/index.html:7]: ?

**Review by sarhan-azizov — CHANGES_REQUESTED** [review]

File: src/components/search/search.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
+import React, { useState } from 'react';
+import PropTypes, { func } from 'prop-types';
+import classNames from 'classnames';
+
+import Icon from '../../icon.js';
+import styles from './search.module.scss';
```
Comment (MozdolevskyiMaksym) [inline src/components/search/search.js:6]: `'./search.scss'`

**Review by MozdolevskyiMaksym — APPROVED** [review]

**Review by sarhan-azizov — APPROVED** [review]

**Pull request merged**